src/build/mdn-comments.ts

```diff
@@ -47,66 +45,124 @@ function extractSummary(markdown: string): string {
     return sentenceMatch[0]; // Return the first full sentence
   }
 
-  return normalizedText.split(" ")[0] || ""; // Fallback: first word if no sentence found
+  const firstWord = normalizedText.split(" ")[0];
+  return firstWord || "";
 }
 
-async function getDirectories(dirPath: URL): Promise<URL[]> {
-  try {
-    const entries = await fs.readdir(dirPath, {
-      withFileTypes: true,
-    });
-    return entries
-      .filter((entry) => entry.isDirectory())
-      .map((entry) => new URL(entry.name + "/", dirPath));
-  } catch (error) {
-    console.error("Error reading directories:", error);
-    return [];
+async function walkDirectory(dir: URL): Promise<URL[]> {
+  const entries = await fs.readdir(dir, { withFileTypes: true });
+  const parentDirName = dir.pathname.split("/").filter(Boolean).slice(-1)[0];
+  let results: URL[] = [];
+
+  for (const entry of entries) {
+    if (entry.isDirectory()) {
+      if (entry.name === parentDirName) continue;
+      const subDir = new URL(`${entry.name}/`, dir);
+      results = results.concat(await walkDirectory(subDir));
+    } else if (entry.isFile() && entry.name === "index.md") {
+      results.push(new URL(entry.name, dir));
+    }
   }
+
+  return results;
 }
 
-async function getIndexMdContents(
-  folders: URL[],
-): Promise<{ [key: string]: string }> {
-  const results: { [key: string]: string } = {};
+const types: Record<string, string> = {
+  property: "properties",
+  method: "methods",
+  "landing-page": "ignore",
+  interface: "interfaces",
+  overview: "ignore",
+  event: "events",
+};
 
-  for (const folder of folders) {
-    const indexPath = new URL("index.md", folder);
+function generateTypes(content: string): string[] | undefined {
+  const pageType = content.match(/page-type:\s*["']?([^"'\n]+)["']?/)!;
+  const type = pageType[1].split("-").pop()!;
+  const plural = types[type];
+  if (!plural) return;
```

Comment:
```suggestion
function generatePath(content: string): string[] | undefined {
  const pageType = content.match(/\npage-type: (.+)\n/)!;
  const type = pageType[1];
  return paths[type];
```

1. This function certainly processes page type but doesn't really return type. Let's call it a path.
2. Splitting by "-" is potentially problematic for `web-api-constructor` as the member `constructor` exists for any object. We could make it a map instead of an object though.